lib/Logger/LogAppender.h

```diff
@@ -77,9 +84,9 @@ class LogAppender {
  
  private:
   static Mutex _appendersLock;
-  static std::vector<std::shared_ptr<LogAppender>> _globalAppenders;
-  static std::map<size_t, std::vector<std::shared_ptr<LogAppender>>> _topics2appenders;
-  static std::map<std::string, std::shared_ptr<LogAppender>> _definition2appenders;
+  static std::unordered_map<std::type_index, std::vector<std::shared_ptr<LogAppender>>> _globalAppenders;
+  static std::unordered_map<std::type_index, std::map<size_t, std::vector<std::shared_ptr<LogAppender>>>> _topics2appenders;
+  static std::unordered_map<std::type_index, std::map<std::string, std::shared_ptr<LogAppender>>> _definition2appenders;
```

Comment:
While using unordered_maps here works and would allow us to have an arbitrary number of log groups, I would prefer if we could use a more lightweight container, such as `std::array`. That would require known in advance how many log groups there will be, but right now we will only have 2. The benefits of using std::array would be that we don't need any typeid magic, no hashing, no memory indirections when finding the appenders for a log group.